cmd/agent/app/reporter/grpc/builder.go

```diff
@@ -90,5 +93,25 @@ func (b *ConnBuilder) CreateConnection(logger *zap.Logger) (*grpc.ClientConn, er
 	}
 	dialOptions = append(dialOptions, grpc.WithDefaultServiceConfig(grpcresolver.GRPCServiceConfig))
 	dialOptions = append(dialOptions, grpc.WithUnaryInterceptor(grpc_retry.UnaryClientInterceptor(grpc_retry.WithMax(b.MaxRetry))))
-	return grpc.Dial(dialTarget, dialOptions...)
+	conn, err := grpc.Dial(dialTarget, dialOptions...)
+
+	if err != nil {
+		return nil, err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	for {
```

Comment:
This will change agent startup to block while it attempts to connect to a collector.  We could run this logic in a goroutine instead, keep the current behavior and just clearly log when a collector connection is successful.

Reply:
+1

Reply:
I would also log "Checking connection to collector" before starting

Reply:
Agree on this.